WordPressKitTests/WordPressOrgXMLRPCApiTests.swift

```diff
@@ -41,4 +42,224 @@ class WordPressOrgXMLRPCApiTests: XCTestCase {
+                // 403 is the 'faultCode' in the HTTP response xml.
+                XCTAssertEqual(error.code, 403)
+                XCTAssertNil(error.userInfo[WordPressOrgXMLRPCApi.WordPressOrgXMLRPCApiErrorKeyData as String])
+                XCTAssertNil(error.userInfo[WordPressOrgXMLRPCApi.WordPressOrgXMLRPCApiErrorKeyStatusCode as String])
+            }
+        )
+        wait(for: [expect], timeout: 0.1)
+    }
+
+    func testMalformedXML() throws {
+        stub(condition: isXmlRpcAPIRequest()) { _ in
+            HTTPStubsResponse(
+                data: #"<?xml version="1.0" encoding="UTF-8"?><methodRespons"#.data(using: .utf8)!,
+                statusCode: 200,
+                headers: ["Content-Type": "application/xml"]
+            )
+        }
+
+        let expect = self.expectation(description: "One callback should be invoked")
+        let api = WordPressOrgXMLRPCApi(endpoint: URL(string: xmlrpcEndpoint)! as URL)
+        api.callMethod(
+            "wp.getPost",
+            parameters: nil,
+            success: { (responseObject: AnyObject, _: HTTPURLResponse?) in
+                expect.fulfill()
+                XCTFail("This call should fail")
+            },
+            failure: { (error, _) in
+                expect.fulfill()
+
+                XCTAssertEqual(error.domain, XMLParser.errorDomain)
+                XCTAssertNotNil(error.userInfo[WordPressOrgXMLRPCApi.WordPressOrgXMLRPCApiErrorKeyData as String])
+                XCTAssertNil(error.userInfo[WordPressOrgXMLRPCApi.WordPressOrgXMLRPCApiErrorKeyStatusCode as String])
+            }
+        )
+        wait(for: [expect], timeout: 0.1)
+    }
+
+    func testInvalidXML() throws {
+        let responseFile = try XCTUnwrap(OHPathForFile("xmlrpc-response-invalid.html", WordPressOrgXMLRPCApiTests.self))
+        stub(condition: isXmlRpcAPIRequest()) { _ in
+            fixture(filePath: responseFile, headers: ["Content-Type": "application/xml"])
+        }
+
+        let expect = self.expectation(description: "One callback should be invoked")
+        let api = WordPressOrgXMLRPCApi(endpoint: URL(string: xmlrpcEndpoint)! as URL)
+        api.callMethod(
+            "wp.getPost",
+            parameters: nil,
+            success: { (responseObject: AnyObject, _: HTTPURLResponse?) in
+                expect.fulfill()
+                XCTFail("This call should fail")
+            },
+            failure: { (error, _) in
+                expect.fulfill()
+
+                XCTAssertEqual(error.domain, WPXMLRPCErrorDomain)
+                XCTAssertEqual(error.code, WPXMLRPCError.invalidInputError.rawValue)
+                XCTAssertNotNil(error.userInfo[WordPressOrgXMLRPCApi.WordPressOrgXMLRPCApiErrorKeyData as String])
+                XCTAssertNil(error.userInfo[WordPressOrgXMLRPCApi.WordPressOrgXMLRPCApiErrorKeyStatusCode as String])
+            }
+        )
+        wait(for: [expect], timeout: 0.1)
```

Comment:
This is totally off topic, but I've been wondering whether it would be worth adopting Nimble here to clear up the code:

```suggestion
        let api = WordPressOrgXMLRPCApi(endpoint: URL(string: xmlrpcEndpoint)! as URL)
        waitUntil { done in
            api.callMethod(
                "wp.getPost",
                parameters: nil,
                success: { (responseObject: AnyObject, _: HTTPURLResponse?) in
                    done()
                    XCTFail("This call should fail")
                },
                failure: { (error, _) in
                    done()

                    XCTAssertEqual(error.domain, WPXMLRPCErrorDomain)
                    XCTAssertEqual(error.code, WPXMLRPCError.invalidInputError.rawValue)
                    XCTAssertNotNil(error.userInfo[WordPressOrgXMLRPCApi.WordPressOrgXMLRPCApiErrorKeyData as String])
                    XCTAssertNil(error.userInfo[WordPressOrgXMLRPCApi.WordPressOrgXMLRPCApiErrorKeyStatusCode as String])
                }
            )
        }
```

🤔 thinking about it, there must be a way to write this kind of code as a function that uses the XCTest APIs internally, which would mean not having to depend on Nimble.

Reply:
It's probably going to be easier to use async test functions than Nimble: There will be even less closures. Considering there will be an async versions of `callMethod` function, I'll level this change to later PRs.